### PR TITLE
LUN-396: Create own voc screen

### PR DESCRIPTION
**Pull request by f1sh1918**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

I spared out the error hints (check design) because i think we don't need them if we disable the save button if required fields are missing.

### CODE REVIEW

**Comment by SteffiStuffel** [comment]
> 
> I spared out the error hints (check design) because i think we don't need them if we disable the save button if required fields are missing.

Designs have changed, and we decided to use the error messages. Have not updated the issue, sorry: 
https://app.zeplin.io/project/5fc7c98a435298126bcfd4ee

File: src/constants/labels.json [hunk 1]
```diff
@@ -180,6 +180,16 @@
...
+    },
+    "creation": {
+      "headline": "Vokabel erstellen",
+      "wordPlaceholder": "Wort eingeben*",
+      "articlePlaceholder": "Artikel auswählen*",
+      "saveButton": "Speichern",
+      "addImage": "Bild hochladen*",
+      "addAudio": "Audio aufnehmen",
+      "maxPictureUpload": "Du kannst bis zu 3 Bilder hochladen",
+      "requiredFields": "*Pflichfelder"
```
Comment (steffenkleinle) [inline src/constants/labels.json:192]:
```suggestion
      "requiredFields": "*Pflichtfelder"
```

File: src/components/Dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import React, { Dispatch, ReactElement, SetStateAction, useState } from 'react'
+import DropDownPicker, { ItemType, ValueType } from 'react-native-dropdown-picker'
+import { widthPercentageToDP as wp } from 'react-native-responsive-screen'
+import styled, { useTheme } from 'styled-components/native'
+
+interface DropdownProps<T> {
```
Comment (steffenkleinle) [inline src/components/Dropdown.tsx:6]: Optional: I think its kind of an convention in our code to put the props type directly above the actual component and not above the styled components. I would prefer to keep it that way.

File: src/components/ScreenHeading.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React, { ReactElement } from 'react'
+import styled from 'styled-components/native'
+
+import { Heading } from './text/Heading'
+
+interface ScreenHeadingProps {
+  text: string
```
Comment (steffenkleinle) [inline src/components/ScreenHeading.tsx:7]: Should be named `title` imo.

File: src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  margin-bottom: ${props => props.theme.spacings.sm};
+`
+
+const StyledHintText = styled(HintText)`
+  margin-left: ${props => props.theme.spacings.xxl};
+`
+
+const UserVocabularyCreationScreen = (): ReactElement => {
+  const [word, setWord] = useState<string>('')
+  const [articleId, setArticleId] = useState<number | null>(null)
```
Comment (steffenkleinle) [inline src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx:44]: Would it make sense to have one article selected as default? Maybe it would make creating own vocabulary a little bit faster if the correct one is already selected. However, the downside may be that user miss that they may have to select another article.

File: package.json [hunk 5]
```diff
@@ -37,6 +37,7 @@
     "react-native": "0.66.4",
     "react-native-camera": "^4.2.1",
     "react-native-device-info": "^9.0.2",
+    "react-native-dropdown-picker": "^5.4.2",
```
Comment (steffenkleinle) [inline package.json:40]: Any reason why you chose that specific library? While its probably fine the last release is 4 months old and there are not that many downloads. This one has a lot more: https://www.npmjs.com/package/@react-native-picker/picker

File: src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        />
+        <Dropdown
+          value={articleId}
+          setValue={setArticleId}
+          placeholder={getLabels().ownVocabulary.creation.articlePlaceholder}
+          items={ARTICLES.filter(article => article.value !== 'keiner') as ArticleType[]}
+          itemKey='id'
+        />
+        <AddPictureButton
+          onPress={() => null}
```
Comment (steffenkleinle) [inline src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx:70]: Will it be done in a separate issue?

File: src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+          clearable
+          value={word}
+          onChangeText={setWord}
+          placeholder={getLabels().ownVocabulary.creation.wordPlaceholder}
+        />
+        <Dropdown
+          value={articleId}
+          setValue={setArticleId}
+          placeholder={getLabels().ownVocabulary.creation.articlePlaceholder}
+          items={ARTICLES.filter(article => article.value !== 'keiner') as ArticleType[]}
```
Comment (steffenkleinle) [inline src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx:66]: Why do we need the typecast?

File: src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  const onSave = (): void => {
+    setWord('')
+    setArticleId(null)
+    // TODO LUN-419
+  }
+
+  return (
+    <RouteWrapper>
+      <Root>
+        <ScreenHeading text={getLabels().ownVocabulary.creation.headline} />
```
Comment (steffenkleinle) [inline src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx:55]: Optional: We could start to destructure these labels:
``` ts
const { heading, wordPlaceholder, articlePlaceholder } = getLabels().ownVocabulary.creation
```

File: src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+          buttonTheme={BUTTONS_THEME.text}
+          iconLeft={AudioCircleIcon}
+          iconSize={wp('10%')}
+        />
+        <HintText>{getLabels().ownVocabulary.creation.requiredFields}</HintText>
+        <SaveButton
+          onPress={onSave}
+          label={getLabels().ownVocabulary.creation.saveButton}
+          buttonTheme={BUTTONS_THEME.contained}
+          disabled={!articleId || word.length === 0}
```
Comment (steffenkleinle) [inline src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx:89]: Maybe
```suggestion
          disabled={!articleId || word.length < 2}
```

**Comment by SteffiStuffel** [comment]
I will test and review, when the error messages are inlcuded.

File: src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx (hunk 4 above)
Comment (f1sh1918) [reply src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx:44]: hm i don't think this is a good idea. what to choose for default?

File: package.json (hunk 5 above)
Comment (f1sh1918) [reply package.json:40]: it was the best fitting library i found. the one you recommended is a picker and not a dropdown and is not fitting to the design if you check the ios version in the screenshots

File: src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx (hunk 6 above)
Comment (f1sh1918) [reply src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx:70]: yes there are separate tickets, should i add a todo?

File: src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx (hunk 7 above)
Comment (f1sh1918) [reply src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx:66]: before the refactoring i needed it to pass to the lib because the type is readonly. but now its obsolete. thx

File: src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx (hunk 9 above)
Comment (f1sh1918) [reply src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx:89]: Hm then we should maybe adjust the error message with sth like (min 3.chars?)

File: src/components/Dropdown.tsx (hunk 2 above)
Comment (f1sh1918) [reply src/components/Dropdown.tsx:6]: didn't know. adjusted

**Comment by f1sh1918** [comment]
> > I spared out the error hints (check design) because i think we don't need them if we disable the save button if required fields are missing.
> 
> Designs have changed, and we decided to use the error messages. Have not updated the issue, sorry: https://app.zeplin.io/project/5fc7c98a435298126bcfd4ee

Alright thx for the hint. Adjusted design. I think the ErrorMessages differ to the other implementations (fontSize). I think we should stick to the other implementations like `AddCustomDisciplineScreen`
So either reduce the `fontSize` everywhere and create `HintError` component or keep it like that

File: src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx (hunk 4 above)
Comment (SteffiStuffel) [reply src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx:41]: Agree with @f1sh1918

File: src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx (hunk 9 above)
Comment (SteffiStuffel) [reply src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx]: i think `=== 0 ` is fine / good enough

File: src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+          value={articleId}
+          setValue={setArticleId}
+          placeholder={articlePlaceholder}
+          items={ARTICLES.filter(article => article.value !== 'keiner')}
+          itemKey='id'
+          errorMessage={articleErrorMessage}
+        />
+        <AddPictureButton
+          onPress={() => null}
+          label={addImage.toUpperCase()}
```
Comment (SteffiStuffel) [inline src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx:97]: No need to toUpperCase() is automatically done for our buttons by default.

File: src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+          buttonTheme={BUTTONS_THEME.text}
+          iconLeft={PhotoCircleIcon}
+          iconSize={wp('10%')}
+        />
+        <StyledHintText>{maxPictureUpload}</StyledHintText>
+        <AddAudioButton
+          onPress={() => null}
+          label={addAudio.toUpperCase()}
```
Comment (SteffiStuffel) [inline src/routes/user-vocabulary/UserVocabularyCreationScreen.tsx:105]: Same as above: No need to toUpperCase() is automatically done for our buttons by default.

**Review by SteffiStuffel — COMMENTED** [review]
Scrollable screen, when the keyboard is opened, would be really nice. But I'm fine, if we do this in a separate ticket, so we can get this merged an get started with audio and image upload. Please create a ticket if you decide to not do it in this PR.

After saving it should redirect to Vocabulary List. I think this should be done in separate ticket, as vocabulary list is not yet available. (I already just created this ticket)


Tested on Android emulator, looks really nice :tada:

File: src/components/PressableOpacity.tsx [hunk 12]
```diff
@@ -10,8 +10,8 @@ interface Props {
   disabled?: boolean
 }
 
-const OPACITY_MIN = 0.2
-const OPACITY_MAX = 1
+export const OPACITY_MIN = 0.2
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply src/components/PressableOpacity.tsx:13]: moved to theme -> styles

File: src/components/ScreenHeading.tsx [hunk 13]
```diff
@@ -0,0 +1,26 @@
+import React, { ReactElement } from 'react'
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply src/components/ScreenHeading.tsx:1]: @SteffiStuffel the spacings are different and the description needs to be optional. I would prefer using title component and make description optional. In some screens more spacing looks better but we could just make `styled(Title)` and adjust margin/padding

File: src/constants/data.ts [hunk 14]
```diff
@@ -75,31 +75,37 @@ export const BUTTONS_THEME = {
 
 export type ButtonTheme = typeof BUTTONS_THEME[keyof typeof BUTTONS_THEME]
 
-interface ArticleType {
+export interface ArticleType {
   readonly id: number
   readonly value: string
+  readonly label: string
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply src/constants/data.ts:81]: alright. i extended the type and created separate function

File: src/constants/labels.json [hunk 15]
```diff
@@ -180,6 +180,17 @@
       "list": "Vokabelliste",
       "create": "Vokabeln erstellen",
       "practice": "Vokabeln üben"
+    },
+    "creation": {
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply src/constants/labels.json:184]: hm i think a noun is fine here

File: src/navigation/NavigationTypes.ts [hunk 16]
```diff
@@ -42,6 +42,9 @@ export type RoutesParams = {
   UserVocabularyTab: undefined
   Home: undefined
   UserVocabularyOverview: undefined
+  UserVocabularyCreation: {
+    title: string
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply src/navigation/NavigationTypes.ts:46]: i think you can reach the creation screen from different screens f.e. "list" and "overview" therefore it makes sense for me to have different titles to label the correct "Page Back"

File: src/navigation/UserVocabularyStackNavigator.tsx [hunk 17]
```diff
@@ -24,6 +25,11 @@ const UserVocabularyStackNavigator = (): JSX.Element | null => {
         component={UserVocabularyOverviewScreen}
         options={({ navigation }) => options(overview, navigation)}
       />
+      <Stack.Screen
+        name='UserVocabularyCreation'
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply src/navigation/UserVocabularyStackNavigator.tsx:29]: Since i would like to put every child screen of the `UserVocabularyOverviewScreen` in a subfolder as `user-vocabulary` i think its better to start with `UserVocabularyXY` so f.e. `UserVocabularyEditScreen`

File: src/components/Dropdown.tsx [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+`
+
+interface DropdownProps<T> {
+  placeholder: string
+  items: ItemType<string | number>[]
+  itemKey: string
+  multiSelect?: boolean
+  setValue: Dispatch<SetStateAction<T | null>>
+  value: T | null
+  schema?: { label: string; value: string }
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply src/components/Dropdown.tsx:27]: you can map your items array with the schema of the dropdown. but probably you are right to keep a default schema since we don't need more than label and value

**Comment by f1sh1918** [comment]
> 

1. I added the KeyboardAwareScrollView --> should be fine now
2. Added TODO with LUN-419 for additional logic

File: ios/Podfile.lock [hunk 19]
```diff
@@ -638,4 +638,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: ca4eb736278d4211ffaa5cef62299ef8fc2137ca
 
-COCOAPODS: 1.11.2
+COCOAPODS: 1.11.3
```
Comment (steffenkleinle) [inline ios/Podfile.lock:641]: Revert

File: src/routes/user-vocabulary/UserVocabularyEditScreen.tsx [hunk 20]
```diff
@@ -0,0 +1,120 @@
...
+
+const StyledHintText = styled(HintText)`
+  margin-left: ${props => props.theme.spacings.xxl};
+`
+
+const UserVocabularyEditScreen = (): ReactElement => {
+  const [word, setWord] = useState<string>('')
+  const [articleId, setArticleId] = useState<number | null>(null)
+  const [wordErrorMessage, setWordErrorMessage] = useState<string>('')
+  const [articleErrorMessage, setArticleErrorMessage] = useState<string>('')
```
Comment (steffenkleinle) [inline src/routes/user-vocabulary/UserVocabularyEditScreen.tsx:44]: Optional: I am still a fan of using `null` as default value here. Makes things a lot more explicit.
```suggestion
  const [wordErrorMessage, setWordErrorMessage] = useState<string | null>(null)
  const [articleErrorMessage, setArticleErrorMessage] = useState<string | null>(null)
```

Comment (steffenkleinle) [reply src/routes/user-vocabulary/UserVocabularyEditScreen.tsx:44]: I think in this case you could also just use booleans since it will only ever be `errorMessage` anyway.

File: src/routes/user-vocabulary/UserVocabularyEditScreen.tsx [hunk 21]
```diff
@@ -0,0 +1,120 @@
...
+    if (!articleId) {
+      setArticleErrorMessage(errorMessage)
+    } else {
+      setArticleErrorMessage('')
+    }
+
+    if (!hasError) {
+      setWord('')
+      setArticleId(null)
+    }
```
Comment (steffenkleinle) [inline src/routes/user-vocabulary/UserVocabularyEditScreen.tsx:74]: Optional: 
```suggestion
   if (word.length > 0 && articleId) {
     setWord('')
     setArticleId(null)
     // TODO LUN-419: Save vocabulary
   }
   setWordErrorMessage(word.length === 0 ? errorMessage : '')
   setArticleErrorMessage(!articleId ? errorMessage : '')
```

File: src/constants/theme/styles.ts [hunk 22]
```diff
@@ -0,0 +1,6 @@
+export const STYLES = {
+  opacity: {
+    min: 0.2,
+    max: 1,
```
Comment (steffenkleinle) [inline src/constants/theme/styles.ts:4]: If these are and will only be used for press opacity of buttons the naming should resemble that imo.

File: src/components/Title.tsx [hunk 23]
```diff
@@ -32,17 +33,22 @@ interface ITitleProps {
...
-    <TitleContent>{description}</TitleContent>
+    {description && <TitleContent>{description}</TitleContent>}
     {children}
   </Container>
 )
 export default Title
+
+export const TitleSpacing = styled(Title)`
+  margin-bottom: ${props => props.theme.spacings.xxl};
+`
```
Comment (steffenkleinle) [inline src/components/Title.tsx:54]: Please move to the top of the file

File: src/constants/data.ts [hunk 24]
```diff
@@ -78,37 +78,43 @@ export type ButtonTheme = typeof BUTTONS_THEME[keyof typeof BUTTONS_THEME]
...
+interface ArticleTypeExtended extends ArticleType {
+  readonly label: string
+}
+
+export const getArticleWithLabel = (): ArticleTypeExtended[] =>
+  ARTICLES.filter(article => article.id !== 0).map(article => {
+    if (article.id === 4) {
+      return { ...article, label: `${article.value} (Plural)` }
+    }
+    return { ...article, label: article.value }
```
Comment (steffenkleinle) [inline src/constants/data.ts:115]:
```suggestion
    return article
```

**Review by steffenkleinle — CHANGES_REQUESTED** [review]
Tested on android, looks nice :writing_hand: 

`Bild hochladen` is marked as required field but there is no check for it. Will it be added in a separate PR or is it marked incorrectly?

Getting `VirtualizedLists should never be nested inside plain ScrollViews` when opening the article dropdown.

File: src/navigation/NavigationTypes.ts (hunk 16 above)
[the remark this answers is not in the document]
Comment (SteffiStuffel) [reply src/navigation/NavigationTypes.ts:46]: Ok, i didn't get it, thought it is about the Title/Heading of the page, a more clear naming woud be nice here. First thought `HeaderBackLabel`?

File: src/navigation/UserVocabularyStackNavigator.tsx (hunk 17 above)
[the remark this answers is not in the document]
Comment (SteffiStuffel) [reply src/navigation/UserVocabularyStackNavigator.tsx]: This not our logic how we structure the routes-folder. We should not change this without talking about it in the team. I would really like if we stick to the existing logic OR have subfolders for every BottomNavigationTab (but we will have exceptions here, as some screens can be in different tabs, e.g. VocabularyDetail).

[the remark this answers is not in the document]
Comment (f1sh1918) [reply src/navigation/UserVocabularyStackNavigator.tsx]: alright. i renamed the Screens to `OverviewUserVocabularyScreen` and `EditUserVocabularyScreen`

File: src/routes/user-vocabulary/UserVocabularyEditScreen.tsx (hunk 20 above)
Comment (f1sh1918) [reply src/routes/user-vocabulary/UserVocabularyEditScreen.tsx:44]: refactored to boolean. thx for the hint

File: src/routes/user-vocabulary/UserVocabularyEditScreen.tsx (hunk 21 above)
Comment (f1sh1918) [reply src/routes/user-vocabulary/UserVocabularyEditScreen.tsx:74]: not needed since refactored to boolean

File: src/constants/data.ts (hunk 24 above)
Comment (f1sh1918) [reply src/constants/data.ts:115]: you have to return a label for `ArticleTypeExtended` so just returning the article is not working

File: src/components/Title.tsx (hunk 23 above)
Comment (f1sh1918) [reply src/components/Title.tsx:54]: since i'm referring to the Title component  itself, i can't move it because i would use it before the declaration of the title component (line 41) -> rejected

**Comment by f1sh1918** [comment]
> Tested on android, looks nice ✍️
> 
> `Bild hochladen` is marked as required field but there is no check for it. Will it be added in a separate PR or is it marked incorrectly?
> 
> Getting `VirtualizedLists should never be nested inside plain ScrollViews` when opening the article dropdown.

1. functionality will be added in separate ticket
2. Since i didn't find an easy solution to fix that issue and keep a keyboard handling i removed the `KeyboardAwareScrollView` and created a separate ticket for it
https://issues.tuerantuer.org/browse/LUN-424

File: src/constants/theme/styles.ts (hunk 22 above)
Comment (f1sh1918) [reply src/constants/theme/styles.ts:4]: renamed `pressOpacity`

File: src/constants/data.ts (hunk 24 above)
Comment (steffenkleinle) [reply src/constants/data.ts:115]: oh yeah sorry :see_no_evil:

**Comment by steffenkleinle** [comment]
@f1sh1918 please resolve merge conflicts and update to current main then I'll review again :)

File: src/navigation/NavigationTypes.ts [hunk 25]
```diff
@@ -42,10 +42,14 @@ export type RoutesParams = {
...
-  UserVocabularyEdit: {
-    title: string
+  OverviewUserVocabulary: undefined
+  EditUserVocabulary: {
+    headerBackLabel: string
   }
+  UserVocabularyList: {
+    headerBackLabel: string
+  }
+  UserVocabularyDetail: { document: Document }
```
Comment (steffenkleinle) [inline src/navigation/NavigationTypes.ts:52]: Naming is now really inconsistent, why did you rename to `OverviewUserVocabulary` and `EditUserVocabulary`? I think we should stick to the previous naming, i.e. `UserVocabularyOverview` and so on.

And what is the difference between overiew and list?

File: src/routes/edit-user-vocabulary/EditUserVocabularyScreen.tsx [hunk 26]
```diff
@@ -0,0 +1,111 @@
+import React, { ReactElement, useState } from 'react'
```
Comment (steffenkleinle) [inline src/routes/edit-user-vocabulary/EditUserVocabularyScreen.tsx:1]: What was changed here? Not reviewed. I really think you should revert renaming.

**Comment by steffenkleinle** [comment]
Is this now also done here? https://issues.tuerantuer.org/browse/LUN-406

File: src/navigation/NavigationTypes.ts (hunk 25 above)
Comment (f1sh1918) [reply src/navigation/NavigationTypes.ts:52]: well then lets talk about naming conventions. i think we had a discussion in this pr.. and `EditUserVocabulary` was the decision so i also changed the Overview Screen name

Comment (f1sh1918) [reply src/navigation/NavigationTypes.ts:52]: Update: i talked to @SteffiStuffel  and we decided to revert this due to some misunderstandings

File: src/routes/edit-user-vocabulary/EditUserVocabularyScreen.tsx (hunk 26 above)
Comment (f1sh1918) [reply src/routes/edit-user-vocabulary/EditUserVocabularyScreen.tsx:1]: renamed

**Review by steffenkleinle — APPROVED** [review]

File: ios/Podfile.lock (hunk 19 above)
Comment (SteffiStuffel) [reply ios/Podfile.lock:641]: @f1sh1918

File: src/routes/user-vocabulary-list/UserVocabularyListScreen.tsx [hunk 27]
```diff
@@ -78,8 +79,8 @@ const UserVocabularyListScreen = ({ navigation }: Props): ReactElement => {
         {!isKeyboardVisible && (
           <ButtonContainer>
             <Button
-              onPress={() => navigation.navigate('UserVocabularyOverview')}
-              label={getLabels().userVocabulary.create}
+              onPress={() => navigation.navigate('UserVocabularyEdit', { headerBackLabel: list })}
```
Comment (SteffiStuffel) [inline src/routes/user-vocabulary-list/UserVocabularyListScreen.tsx:82]: hmh, `edit` is not really a good naming for "edit" or "create". I do not have a good idea though.

**Review by SteffiStuffel — COMMENTED** [review]
Tested, looks good.

File: src/components/Title.tsx [hunk 28]
```diff
@@ -32,17 +33,22 @@ interface ITitleProps {
...
     <TitleHeading>{title}</TitleHeading>
     {subtitle && <TitleSubheading>{subtitle}</TitleSubheading>}
-    <TitleContent>{description}</TitleContent>
+    {description && <TitleContent>{description}</TitleContent>}
     {children}
   </Container>
 )
 export default Title
+
+export const TitleSpacing = styled(Title)`
```
[the remark this answers is not in the document]
Comment (steffenkleinle) [reply src/components/Title.tsx:52]: agree with @SteffiStuffel

File: src/components/SearchBar.tsx [hunk 29]
```diff
@@ -14,7 +14,7 @@ const SearchBar = ({ query, setQuery }: Props): ReactElement => (
   <CustomTextInput
     value={query}
     onChangeText={setQuery}
-    placeholder={getLabels().search.enterWord}
+    placeholder={getLabels().dictionary.enterWord}
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply src/components/SearchBar.tsx:17]: well then it makes more sense to pass a var into the searchBar component called `placeholder` @SteffiStuffel 
If not every `SearchBar` got the same placeholder and i don't understand your comment :)

[the remark this answers is not in the document]
Comment (SteffiStuffel) [reply src/components/SearchBar.tsx:17]: Sorry for being unclear:
- Every search bar has the same placeholder.
- SeachBar-component can be used, e.g. in DictionaryScreen, UserVocabularyListScreen, FavoritesScreen.
- So naming the label/constant to "dictionary" does not make sense, as it is not only in the DicitionaryScreen. Label should be called `getLabels().search.enterWord` as it was previously.

File: ios/Podfile.lock (hunk 19 above)
Comment (f1sh1918) [reply ios/Podfile.lock:641]: sorry for that. Very annoying...created a ticket for that and reverted podfile in last commit
https://issues.tuerantuer.org/browse/LUN-437

File: src/components/SearchBar.tsx (hunk 29 above)
[the remark this answers is not in the document]
Comment (f1sh1918) [reply src/components/SearchBar.tsx:17]: well i reverted it

File: src/routes/user-vocabulary-list/UserVocabularyListScreen.tsx (hunk 27 above)
Comment (f1sh1918) [reply src/routes/user-vocabulary-list/UserVocabularyListScreen.tsx:82]: what about `process` ? i really don't want to block this pr due to naming ;)

Comment (SteffiStuffel) [reply src/routes/user-vocabulary-list/UserVocabularyListScreen.tsx:130]: Fine for me. Yes let's get this merged soon, can change naming afterwards, if needed.

Comment (f1sh1918) [reply src/routes/user-vocabulary-list/UserVocabularyListScreen.tsx]: renamed.

**Review by SteffiStuffel — APPROVED** [review]
